Mathlib/RingTheory/Nilpotent/Lemmas.lean

```diff
@@ -111,6 +111,12 @@ end LinearMap
 
 namespace Module.End
 
+lemma isNilpotent_restrict {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M]
```

Comment:
```suggestion
lemma IsNilpotent.restrict {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M]
```
for max dot notation (-;
Shouldn't `hf` be explicit?

Reply:
I agree `hf` should be explicit and I have renamed to `IsNilpotent.restrict`.

However I note that we still cannot use dot notation because this is in `Module.End` and not the root namespace, and I am not convinced it should be in the root.